e2e/app/page/google-login.ts

```diff
@@ -49,56 +48,86 @@ export default class GoogleLoginPage {
+      if (link) {
+        const isLink = await this.page.evaluate((e1, e2) => e1 === e2, elemt1, link);
+        if (isLink) {
+          // click "Use another Account" link and wait for navigation.
+          await link.click();
+        }
+        await link.dispose();
+      }
+
+      const emailInput = await this.email();
+      await emailInput.focus();
+      await emailInput.type(userEmail, {delay: 15});
+      await emailInput.dispose();
+
+      const nextButton = await this.page.waitForXPath(FieldSelector.NextButton, {visible: true});
+      await nextButton.click();
+      await nextButton.dispose();
+    } catch(error) {
+      await takeScreenshot(this.page);
+      await savePageToFile(this.page);
+      throw error;
+    }
   }
 
   /**
    * Enter login password.
    * @param pwd
    */
   async enterPassword(pwd: string) : Promise<void> {
-    const input = await this.password();
-    await input.focus();
-    await input.type(pwd, {delay: 15});
-    await input.dispose();
+    try {
+      const input = await this.password();
+      await input.focus();
+      await input.type(pwd, {delay: 15});
+      await input.dispose();
+    } catch (error) {
+      await takeScreenshot(this.page);
+      await savePageToFile(this.page);
+      throw error;
+    }
   }
 
   /**
    * Click Next button to submit login credential.
    */
-  async submit() : Promise<void> {
-    const submitButton = new Button(this.page, FieldSelector.SubmitButton);
-    const [response] = await Promise.all([
-      this.page.waitForNavigation({waitUntil: ['networkidle2', 'load'], timeout: 180000}),
-      submitButton.click(),
-    ]);
-    const status = response.status();
-    if (status !== 200) {
-      console.error(`Login navigation response status: ${status}`);
+  async submit(): Promise<void> {
+    try {
+      const submitButton = new Button(this.page, FieldSelector.SubmitButton);
+      await Promise.all([
+        this.page.waitForNavigation({waitUntil: ['networkidle2', 'load'], timeout: 60000}),
+        submitButton.click()
+      ]);
+      await submitButton.dispose();
+    } catch (err) {
+      // If navigation errored, it could be because prompted with "Enter Recovery Email" page.
+      const recoverEmailXpath = '//input[@type="email" and @aria-label="Enter recovery email address"]';
+      await Promise.race([
+        this.page.waitForSelector('app-signed-in'),
+        this.page.waitForXPath(recoverEmailXpath, {visible: true})
+      ]);
+      const [elementHandle] = await this.page.$x(recoverEmailXpath);
+      if (elementHandle) {
+        await elementHandle.type(config.institutionContactEmail);
+        await Promise.all([
+          this.page.waitForNavigation({waitUntil: ['networkidle2', 'load'], timeout: 60000}),
+          this.page.keyboard.press(String.fromCharCode(13)), // press Enter key
+          this.page.waitForSelector('app-signed-in')
+        ]);
+        await elementHandle.dispose();
+      }
```

Comment:
Should we be doing something here if `elementHandle` does not exist? It seems like we leave the potential for `err` to be swallowed by this catch block. That's usually an anti-pattern, but I can't tell from the context whether that's desired behavior here. If it is, it's worth a comment.